### PR TITLE
Set up CI with Azure Pipelines

### PR DESCRIPTION
**Pull request by Edilmo**
This PR contains the minimal CI required to work with this repo in Azure DevOps.
The main idea behind this initial step is the following: let's start with a CI as closer as possible to the CI in upstream, and improve or adapt on top of that taking advantage of all the features that Azure Pipelines offer.

Several task has been deferred:
- Public PR to move to Bazel 2.0: the two files that are modified out of the folder ci/azure-pipelines, are related to the support of Bazel 2.0, which is the version used by the CI added by this PR.
- Streaming related tests are not integrated yet.
- Bazel Remote Cache.
- Parallelization of jobs: currently there is one very long job that concentrates all the integration and performance tests of RLlib, Tune and others. This could be easily split in the near future.
- Performance considerations in Microsoft-hosted machines: there is a small change in a threshold that was necessary because a random sampling implementation was performing a little bit poor in Microsoft-hosted Ubuntu.
- Tune related tests are disabled: there is incompatibility issue with PyTorch.
- Large memory tests are disabled: the current pool has not enough memory.
- And more...

The missing parts mention above and others will be added in incremental fashion.

### CODE REVIEW

File: ci/azure_pipelines/templates/artifacts.yml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+steps:
+- task: ComponentGovernanceComponentDetection@0
+  inputs:
+    scanType: 'Register'
+    verbosity: 'Verbose'
+    # dockerImagesToScan: 'docker-image'
+    alertWarningLevel: 'Critical'
+    failOnAlert: true
+    # ignoreDirectories: '**/_bazel_runner'
```
Comment (NicholasGorski) [inline ci/azure_pipelines/templates/artifacts.yml]: Intended to be commented out? (Or perhaps meant to ignore all bazel-* directories?)

File: ci/azure_pipelines/templates/info.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    # the hash was computed in Mac
+    if [[ $AGENT_OS == "Darwin" ]]; then 
+      pushd $BUILD_SOURCESDIRECTORY
+      EXPECTED_HASH_TRAVIS='f98de9357fe7e679c7e0522ba84bbfd7a959b7b0d368427630d3ccd0bffd3097'
+      CURRENT_HASH_TRAVIS=$(shasum -a 256 ./.travis.yml | awk '{print $1}')
+      if [[ $EXPECTED_HASH_TRAVIS != $CURRENT_HASH_TRAVIS ]]; then 
+        echo "The original Travis file of the project has changed"
+        echo "Expected Hash: $EXPECTED_HASH_TRAVIS"
+        echo "Current Hash:  $CURRENT_HASH_TRAVIS"
+        echo "Please check the changes, change the azure pipelines acordingly and update the sha256"
```
Comment (NicholasGorski) [inline ci/azure_pipelines/templates/info.yml]: Nit: accordingly* here and below

File: ci/azure_pipelines/templates/artifacts.yml (hunk 1 above)
Comment (Edilmo) [reply ci/azure_pipelines/templates/artifacts.yml]: Good catch. I forgot this.

The first comment line about docker, it's just a place holder to know where to configure docker images that have to be reviewed by the governance task. I haven't identified one yet. But given the fact that this is not documented and it was tricky to find the options, I will put this commented out but with empty value.

Regarding the other line, I'm not 100% sure what is the right thing to do here: if we should ignore everything that Bazel is pulling in order to build faster or let Component Governance be aware of everything. I chose the last. But I will do the same that above: comment out and put empty value.

Thx

File: ci/azure_pipelines/templates/info.yml (hunk 2 above)
Comment (Edilmo) [reply ci/azure_pipelines/templates/info.yml]: Done!
I also addressed all the new lines that were missing.
Thx

**Pull request merged**